File: OneDrive/Desktop/COEXIST/backcoexist-final1SpringBoot/src/main/java/tn/esprit/coexist/entity/Carpooling.java
package tn.esprit.coexist.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.time.LocalTime;
import java.util.Date;
import java.util.List;
import java.util.Timer;

@Entity
@Data

public class Carpooling {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer carpoolingID;

    private LocalDateTime departureTime;
    private String longitudeDeparture;
    private String latitudeDestination;

    private String latitudeDeparture;
    private String longitudeDestination;



    private Integer availableSeats;
    private float costPerSeat;
    @Enumerated(EnumType.STRING)
    private Day day;
    LocalTime time;

    @Enumerated(EnumType.STRING)
    private CarpoolingType carpoolingType;
    private Long registrationNumber;
    @ManyToOne
    @JsonIgnore
    private User user;
    @OneToMany(mappedBy ="carpooling",cascade = CascadeType.ALL)

    private List<Booking>bookings;
    @Override
    public String toString() {
        return "Carpooling{" +
                "carpoolingID=" + carpoolingID +
                ", departureTime=" + departureTime +
                ", user=" + user + // Include other fields as needed
                '}';
    }
}

File: OneDrive/Desktop/COEXIST/FrontCoexist-final1Angular/FrontCoexist-final1/src/app/entity/Carpooling.ts
// Carpooling.ts
import {CarpoolingType} from "./CarpoolingType";
import {Day} from "./Day";
import {Booking} from "./Booking";
import {UserModel} from "./User.model";

export class Carpooling {
  carpoolingID!: number;
  departureTime!: string;
  longitudeDeparture!: string;
  latitudeDeparture!: string;
  longitudeDestination!: string;
  latitudeDestination!: string;
  availableSeats!: number;
  costPerSeat!: number;
  day!: Day; // Enum type
  time!:String;
  carpoolingType!: CarpoolingType; // Enum type
  registrationNumber!: number;
  bookings!: Booking[];


  user!: UserModel; // Assuming you have a User class
 // Assuming you have a Booking class
}
